server/item/writable_book.go

```diff
@@ -0,0 +1,100 @@
+package item
+
+type WritableBook struct {
+	Pages []string
+}
+
+// Page returns a specific page from the book. If the page exists, it will return the content and true, otherwise
+// it will return an empty string and false.
+func (w WritableBook) Page(page uint) (string, bool) {
+	p := int(page)
+	if len(w.Pages) <= p {
+		return "", false
+	}
+	return w.Pages[p], true
+}
+
+// MaxCount always returns 1.
+func (w WritableBook) MaxCount() int {
+	return 1
+}
+
+// Exists checks to see weather a page exists or not
+func (w WritableBook) Exists(page uint) bool {
+	if len(w.Pages) <= int(page) {
+		return false
+	}
+	return true
+}
+
+// Set writes a page to the book, if the page doesn't exist it will be created. It will panic if the
+// text is longer then 256 characters. It will return the newly updated pages.
+func (w WritableBook) Set(page uint, text string) []string {
+	if len(text) > 256 {
+		panic("text longer then 256 bytes")
+	}
```

Comment:
Logic here only allows 255 bytes, and the error message says the text is longer than 256 which would be incorrect if 256 bytes were provided

Reply:
Logic here allows for 256, only panics once its above 256. Unless I'm somehow blanking on how the len() method works.